### PR TITLE
chore(release): added semantic release to automate release

### PR DESCRIPTION
**Pull request by nellyk**
## Description

This change introduces `semantic-release` https://semantic-release.gitbook.io automating the building  and publishing of one-app docker images,generating release notes, adding a changelog, creating a new version depending on the commit messages and publishing github releases.

Since we are currently working on a prerelease in order to continue using this we would have to switch to working from a beta branch so that semantic release can trigger prereleases. Once we are ready to release a major version we can merge to master and that would trigger the initial release

To see this in action please checkout this out

github - https://github.com/nellyk/one-app/commits?author=semantic-release-bot
travis - https://travis-ci.org/nellyk/one-app/builds/658765417?utm_medium=notification&utm_source=github_status
docker - https://hub.docker.com/repository/docker/nkiboi/one-app
## Motivation and Context
 This would automate the release process for one-app and publish docker images to the docker registry
 This enables us to automate the release process which currently is not the case 

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality) Add semantic release for automating builds
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation) 

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly. 
- [x] These changes should be applied to a maintenance branch. Creating a new branch to cater for prerelease
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?

This would allow developer to be up to date with releases and consume the official docker images for one-app

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -16,4 +16,10 @@ jobs:
...
       script: npm run test:lint && npm run test:unit && npm run test:danger && npm run test:git-history
     - name: "Integration Tests"
-      script: ./scripts/run-integration-tests.travis.sh
\ No newline at end of file
+      script: ./scripts/run-integration-tests.travis.sh
+    - stage: release
+      if: branch = master
+      script:
+       - export REPO=oneamex/one-app
+       - docker build -f Dockerfile -t $REPO .
```
Comment (anescobar1991) [inline .travis.yml]: we need to build and publish both the dev image (that one app runner and such tooling will use for local dev and tests) as well as the production image.

https://github.com/americanexpress/one-app/blob/master/Dockerfile#L31 is what we use to build the dev image.

File: .travis.yml [hunk 2]
```diff
@@ -16,4 +16,10 @@ jobs:
       name: "Unit Tests"
       script: npm run test:lint && npm run test:unit && npm run test:danger && npm run test:git-history
     - name: "Integration Tests"
-      script: ./scripts/run-integration-tests.travis.sh
\ No newline at end of file
+      script: ./scripts/run-integration-tests.travis.sh
+    - stage: release
+      if: branch = master
+      script:
```
Comment (anescobar1991) [inline .travis.yml]: Do we have a task to publish the One App static assets to a CDN? Could potentially `npm publish` the statics only in order to leverage https://www.jsdelivr.com

File: .travis.yml [hunk 3]
```diff
@@ -16,4 +16,10 @@ jobs:
       name: "Unit Tests"
       script: npm run test:lint && npm run test:unit && npm run test:danger && npm run test:git-history
     - name: "Integration Tests"
-      script: ./scripts/run-integration-tests.travis.sh
\ No newline at end of file
+      script: ./scripts/run-integration-tests.travis.sh
+    - stage: release
```
Comment (anescobar1991) [inline .travis.yml]: I don't think we should do this on every commit to master, we should do this on a new git tag or a similar process but I don't think we intend to release a new One App version on every new commit in master.

**Comment by nellyk** [comment]
closing this as we would like to revisit this process.

**Pull request closed without merging**